### PR TITLE
Hotfix/osg modeling as a library

### PR DESCRIPTION
**Pull request by kapecp**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -364,6 +364,31 @@ if( USE_BUILTIN_DEPENDENCIES )
 		endif()
 	endif() # OpenCV_FOUND
 
+
+	# Build osgModeling
+	ExternalProject_Add( dep_osgModeling
+		PREFIX aruco
```
Comment (IceHax) [inline CMakeLists.txt]: myslim ze tu by mal byt
PREFIX osgModeling

Comment (kapecp) [inline CMakeLists.txt]: spravne, ma tam byt osgModeling - mozete to opravit

**Comment by kapecp** [comment]
pre spravne skompilovanie pod Win mozno bude treba:
- pod riadok 376 pridat -DOSGMODELING_LIBRARY
- zakomentovat riadok 283

**Comment by IceHax** [comment]
Cmake prejde ale pri builde da chybu:

Parse error in command line argument: -DOSGMODELING_LIBRARY
Should be: VAR:type=value
CMake Error: No cmake script provided.
CMake Error: Problem processing arguments. Aborting.

netreba na Win namiesto
ADD_DEFINITIONS(-DOSG_LIBRARY_STATIC)
dat
ADD_DEFINITIONS(-DOSGMODELING_LIBRARY)

(skusal som ale ajtak to nejde)

**Comment by kapecp** [comment]
skuste pod riadok 376 pridat -DOSGMODELING_LIBRARY=TRUE alebo -DOSGMODELING_LIBRARY=ON

ADD_DEFINITIONS(-DOSG_LIBRARY_STATIC) by sa mal zakomentovat

**Comment by IceHax** [comment]
TRUE aj ON prejdu ale asi to nic nerobi, nasledovna chyba buildu bola aj pred zakomentovanim riadku 283 a pridanim pod riadok 376

D:\git\3dsoftviz\dependencies\osgModeling\src\osgModeling\NormalVisitor.cpp:196: error: C2039: 'setNormalIndices' : is not a member of 'osg::Geometry'
D:\OpenSceneGraph-3.2.1\install\include\osg\Geometry:29: see declaration of 'osg::Geometry'
D:\git\3dsoftviz\dependencies\osgModeling\src\osgModeling\NormalVisitor.cpp:196: error: C2039: 'getVertexIndices' : is not a member of 'osg::Geometry'
D:\OpenSceneGraph-3.2.1\install\include\osg\Geometry:29: see declaration of 'osg::Geometry'

**Comment by IceHax** [comment]
-DOSGMODELING_LIBRARY:BOOL=TRUE alebo -DOSGMODELING_LIBRARY:BOOL=ON
rovnaky vysledok

**Comment by kapecp** [comment]
dost velky problem - hladam riesenie

**Comment by kapecp** [comment]
predpokladam ze pouzivate OSG 3.2.x alebo 3.4.x, ze?

**Comment by IceHax** [comment]
Ano, verzia 3.2.1

**Comment by kapecp** [comment]
opravil som osgModeling - treba si checkout-nut v dependencies/osgModeling brach hotfix/deprecated_osg-Geometry
mohlo by to vyriesit tie errory vyssie (mozu sa vsak obajvit ine)

**Comment by IceHax** [comment]
Nasiel som tam drobnu chybu
https://github.com/kapecp/osgmodeling/commit/67711ca3a1667790caae4f319454cecb215dbd60#commitcomment-14457417

Myslim ze sme sa pohli dalej, momentalna chyba:

D:\git\3dsoftviz\dependencies\osgModeling\src\osgModeling\NormalVisitor.cpp:196: error: C2664: 'deprecated_osg::Geometry::setNormalIndices' : cannot convert parameter 1 from 'const osg::IndexArray *' to 'osg::IndexArray *'
Conversion loses qualifiers

ak tam pouzijem const_cast tak dalsia chyba je uz linker
:-1 : error: LNK1181: cannot open input file 'osg.lib'

CMake mam v stave 
283 #ADD_DEFINITIONS(-DOSG_LIBRARY_STATIC)
370 PREFIX osgModeling
377 -DOSGMODELING_LIBRARY:BOOL=TRUE

**Comment by kapecp** [comment]
OK, const_cast moze byt riesenie
pre cannot open input file 'osg.lib' potrebujem obsah suboru
_build/osgModeling/src/osgModeling/CMakeFiles/osgModeling.dir/link.txt
alebo vypis buildu s VERBOSE=1 (iba koniec s prikazom pr cl.exe)

**Comment by kapecp** [comment]
pripadne este skuste v hlavnom CMakeLists.txt zamenit riadok
-DCMAKE_BUILD_TYPE:STRING=Release
za 
-DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}

**Comment by IceHax** [comment]
tu je build log
https://gist.github.com/IceHax/98ee236d2cee3e5b1e72

s
-DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
hlada osgd.lib (debug build)

**Comment by IceHax** [comment]
je tam vidiet
 OPENSCENEGRAPH_LIB_DIR-NOTFOUND

skusil som definovat env var
 OPENSCENEGRAPH_LIB_DIR = D:\OpenSceneGraph-3.2.1\install\lib

ale nepomohlo

**Comment by IceHax** [comment]
vcerajsi compile log
https://gist.github.com/IceHax/8c1e43a665cde159fc3e

**Pull request merged**